### PR TITLE
Upgrade to go 1.17

### LINKED ISSUE #1300: Update to go 1.16
### Description
We should consider upgrading to go1.16 (if not further), to resolve a few vulnerabilities. 

Relevant slack [here](https://buildpacks.slack.com/archives/CD61YAG69/p1632253379043000)

### Additional context
We should check with our library consumers (e.g. Skaffold), to ensure that there won't be any compatibility issues with this.

### PR DESCRIPTION
**Pull request by jromero**
## Related

Resolves #1300

### CODE REVIEW

File: acceptance/assertions/image.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build acceptance
 // +build acceptance
```
Comment (dfreilich) [inline acceptance/assertions/image.go:2]: I see from this that we need to have both the `go:build` and `+build` lines: https://go.googlesource.com/proposal/+/master/design/draft-gobuild.md

File: internal/sshdialer/ssh_dialer_test.go [hunk 2]
```diff
@@ -506,9 +506,10 @@ func prepareSSHServer(t *testing.T) (connConfig *ConnectionConfig, cleanUp func(
 		{HostIP: connConfig.hostIPv4},
 	}
 
-	// lcow doesn't support ipv6 port bindings
+	// docker desktop doesn't support ipv6 port bindings
```
Comment (dfreilich) [inline internal/sshdialer/ssh_dialer_test.go:509]: We'll need to rebase on main for these

**Review by dfreilich — APPROVED** [review]
Super thorough, with upgrading the `tools` and all of the images. Good job!

File: internal/sshdialer/ssh_dialer_test.go (hunk 2 above)
Comment (jromero) [reply internal/sshdialer/ssh_dialer_test.go]: Done as per https://github.com/buildpacks/pack/pull/1317

File: acceptance/assertions/image.go (hunk 1 above)
Comment (jromero) [reply acceptance/assertions/image.go:2]: Correct. It's worth noting that `go fmt` on 1.17 does this for us automatically.

See https://golang.org/doc/go1.17#go-command

**Comment by jromero** [comment]
Strange but on WCOW the tests are failing because the run image isn't created/found when running acceptance tests.

```text
=== CONT  TestAcceptance
    acceptance_test.go:2761: expected nil: Error response from daemon: No such image: pack-test/run:latest
=== CONT  TestAcceptance/acceptance_suite/p_current_cb_current_lc_default/stack_is_created/builder_is_created/complex_builder/builder_has_duplicate_buildpacks/buildpack_layers_have_no_duplication
    testing.go:1169: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
=== CONT  TestAcceptance/acceptance_suite
    testing.go:1169: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
```

**Pull request merged**